### DISCUSSION TITLE
How to use Bookmarklets?

### DISCUSSION THREAD

**Discussion by cheater** [YoshifumiFuyuno/Search-from-Popup-or-ContextMenu#46] [Q&A]
I heard the addon can use bookmarklets. There is some code under these two urls:

https://github.com/YoshifumiFuyuno/Search-from-Popup-or-ContextMenu/wiki/Example-engines
https://github.com/YoshifumiFuyuno/Search-from-Popup-or-ContextMenu/wiki/Auto-execute-Bookmarklet

however, it is not clear how to use them with SPC.

The "Example engines" wiki page has some links, where there is a JSON object in the link anchor (the stuff after the `#`). I don't know how to use it. When I click on it, the link disappears and becomes normal text.

What are the steps necessary to use a bookmarklet that's found on the Example engines page, under Firefox 97?

**Comment by cheater** [comment]
I see that clicking the link added the search engine to SPC in SPC options.

What if there's now link? How to add the code then?

I see that in the options, when you add an engine, you can select the type to "GET", "POST", or "Bookmarklet". Is the correct way to select "Bookmarklet" and just paste the code in there?

**Comment by YoshifumiFuyuno** [comment]
that's right.
If you want to use the code you have written for your own personal use
Select `Bookmarklet` from the type when you add the engine, and write or paste the code in the text area.
